src/views/esp/view_esp.cpp

```diff
@@ -85,6 +85,85 @@ namespace big
 			if (distance < g.esp.box_render_distance[1] && distance > g.esp.box_render_distance[0] && g.esp.box)
 				draw_list->AddRect({esp_x - (62.5f * multplr), esp_y - (175.f * multplr)}, {esp_x - (62.5f * multplr) + (125.f * multplr), esp_y - (175.f * multplr) + (350.f * multplr)}, esp_color);
 
+			if (distance < g.esp.skeleton_render_distance[1] && distance > g.esp.skeleton_render_distance[0] && g.esp.skeleton)
+			{
+				std::vector<PedBones> bone_ids = // size: 18
+				{
+					PedBones::SKEL_Head, 
+					PedBones::SKEL_Neck_1,
+					PedBones::SKEL_Spine_Root, 
+					PedBones::SKEL_Pelvis, 
+					PedBones::SKEL_L_Thigh, 
+					PedBones::SKEL_R_Thigh, 
+					PedBones::SKEL_L_Calf, 
+					PedBones::SKEL_R_Calf, 
+					PedBones::SKEL_L_Foot, 
+					PedBones::SKEL_R_Foot, 
+					PedBones::SKEL_L_Clavicle, 
+					PedBones::SKEL_R_Clavicle, 
+					PedBones::SKEL_L_UpperArm, 
+					PedBones::SKEL_R_UpperArm, 
+					PedBones::SKEL_L_Forearm, 
+					PedBones::SKEL_R_Forearm, 
+					PedBones::SKEL_L_Hand, 
+					PedBones::SKEL_R_Hand
+				};
+
+				std::vector<rage::fvector4> bones = {};
+				const size_t skel_size            = 18;
+
+				for (int i = 0; i < skel_size; i++)
+				{
+					rage::fvector4 out = {};
+					if (g_pointers->m_gta.m_get_ped_bone(plyr->get_ped(), out, bone_ids[i]))
+					{
+						bones.push_back(out);
+					}
+				}
+
+				if (bones.size() == skel_size)
```

Comment:
This check is erroneous. Instead, do a model check or something.

Reply:
why is it wrong lol? it works, just to check if all the bones got added, if one didn't make it through for some reason then the size would be below 18, returning false.

You're the one who added the sig, if its hardcoded to always return true then yes this check is useless, haven't taken a look at it in IDA.

Reply:
We are getting crashes from get_ped_bone, and this might be too late in the call chain to deal with it. 

See #2433 (@maybegreat48 personally reported it to me too, which lead to) and #2431. We can't figure out why they were crashing, and if you add this, you'll probably just resurrect the problem for people who have skeletin mode on. You could surround the call in a try catch and swallow the exception, but that's a terrible solution in my book. Ideally, we should create a thread-safe way to call the natives from the DX layer like YimMenuv2 has, maybegreat suggested we port it over so we don't have to deal with this shit, but I digress, If this scenario occurs, it will be too late in the call chain to deal with it.

Reply:
I'll take a look at it, using a signature to a function that's a native shouldn't create any threading issues though. I've never experienced that personally when I experimented with those threading related stuff.

Reply:
Solved it, @gir489returns

Switching byte 60 to ? on the sig seemed to do the trick.

![fixed_skelly](https://github.com/YimMenu/YimMenu/assets/149878244/2d1be49c-b810-44d9-ad0e-de3272df17a5)